core/src/subgraph/instance_manager.rs

```diff
@@ -1268,3 +1296,125 @@ fn persist_dynamic_data_sources<T: RuntimeHostBuilder<C>, C: Blockchain>(
     // Merge filters from data sources into the block stream builder
     ctx.state.filter.extend(data_sources.iter());
 }
+
+/// Returns true if the timer has elapsed given duration/threshold.
+fn has_threshold_passed(timer: Instant, threshold: Duration) -> bool {
+    timer.elapsed() >= threshold
+}
```

Comment:
Isn't the timer ever reset?

Reply:
I had the reset, but I removed because it didn't seem needed, but now thinking again, the way it is right now it will try to check if it's synced for every processed block, which isn't ideal/efficient.

I'll change this, thanks for pointing out!

Reply:
I just reverted the commit that does this 😊